erasure_test.go

```diff
@@ -0,0 +1,231 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+// mustEncodeData - encodes data slice and provides encoded 2 dimensional slice.
+func mustEncodeData(data []byte, dataBlocks, parityBlocks int) [][]byte {
+	encodedData, err := encodeData(data, dataBlocks, parityBlocks)
+	if err != nil {
+		// Upon failure panic this function.
+		panic(err)
+	}
+	return encodedData
+}
+
+// Generates good encoded data with one parity block and data block missing.
+var getGoodEncodedData = func(data []byte, dataBlocks, parityBlocks int) [][]byte {
+	encodedData := mustEncodeData(data, dataBlocks, parityBlocks)
+	encodedData[7] = nil
+	encodedData[1] = nil
+	return encodedData
+}
+
+// Generates bad encoded data with one parity block and data block with garbage data.
+var getBadEncodedData = func(data []byte, dataBlocks, parityBlocks int) [][]byte {
+	encodedData := mustEncodeData(data, dataBlocks, parityBlocks)
+	encodedData[7] = []byte("garbage")
+	encodedData[1] = []byte("garbage")
+	return encodedData
+}
+
+// Generates encoded data with all data blocks missing.
+var getMissingData = func(data []byte, dataBlocks, parityBlocks int) [][]byte {
+	encodedData := mustEncodeData(data, dataBlocks, parityBlocks)
+	for i := 0; i < dataBlocks+1; i++ {
+		encodedData[i] = nil
+	}
+	return encodedData
+}
+
+// Generates encoded data with less number of blocks than expected data blocks.
+var getInsufficientData = func(data []byte, dataBlocks, parityBlocks int) [][]byte {
+	encodedData := mustEncodeData(data, dataBlocks, parityBlocks)
+	// Return half of the data.
+	return encodedData[:dataBlocks/2]
+}
+
+// TestErasureEncode checks for encoding for different data sets.
+func TestErasureEncode(t *testing.T) {
+	data := []byte("Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.")
+
+	// List of data blocks the test will run on.
+	var listDataBlocks = []int{
+		4,
```

Comment:
Why was 1 data block and 4 parity chosen, but the comment says test will run on data blocks?